server/src/world/entity_scope_map.rs

```diff
@@ -20,13 +20,18 @@ impl<E: Copy + Eq + Hash> EntityScopeMap<E> {
         }
     }
 
-    pub fn get(&self, user_key: &UserKey, entity: &E) -> Option<&bool> {
+    pub fn get(&self, user_key: &UserKey, entity: &E) -> bool {
         let key = (*user_key, *entity);
 
-        self.main_map.get(&key)
+        self.main_map.get(&key).map(|k| *k).unwrap_or(false)
     }
 
     pub fn insert(&mut self, user_key: UserKey, entity: E, in_scope: bool) {
+        if self.main_map.contains_key(&(user_key, entity)) {
```

Comment:
Added in this check to see if the scope is just being updated rather than inserted.
If the user/entity key already existed in the main_map, then we should be able to assume that we just need to update main_map to the new boolean value